### DISCUSSION TITLE
To invoke SU2 from c++

### DISCUSSION THREAD

**Discussion by dlyltm** [su2code/SU2#2287] [Q&A]
Is there API in SU2? I want to invoke SU2's API(if exits) or function from c++ in vs under Windows, how to do it?

**Comment by pcarruscag** [comment]
Everything you can do with the python wrapper exists in c++

**Comment by dlyltm** [comment]
OK, thanks. But I don't know how to use python.

**Reply by bigfooted** [reply]
SU2 is written in C++, so you if you are writing something in C++, you can just include/call/link whatever stuff from SU2 you need to your C++ code. Only if you want to call SU2 functions from another programming language you need an API, like the python API, aka the python wrapper. 
So: 
1. you write something in C++? -> Just use the SU2 include files
2. you write something in python -> use the python API (we have pysu2)

**Reply by dlyltm** [reply]
Thanks. Can I call SU2's code with c++ in VS?

**Reply by bigfooted** [reply]
Yes, you can call SU2 code with C++, because it *is* C++.

**Reply by dlyltm** [reply]
Is there a guide to API or functions of SU2? I'm sorry I couldn't find it.
